setup.py

```diff
@@ -294,6 +294,7 @@ def finalize_options(self):
             extra_compile_args=CFLAGS,
             extra_link_args=LDFLAGS),
     ],
+    install_required=["typing_extensions>=3.7.4.3"],
```

Comment:
May I suggest not installing `typing_extensions` and instead using `typing` module if Python is 3.8 and newer?
```suggestion
    install_required=['typing-extensions;python_version<"3.8"']
```

Since `Protocol`, `Literal` and `Final` were added to typing module since 3.8 you can do something like this:
```python
import sys
if sys.version_info >= (3, 8):
    from typing import Final, Literal, Protocol
else:
    from typing_extensions import Final, Literal, Protocol
```

Reply:
That's a good catch. I'm going to do the `sys.version_info` check in `compat.py` and then use `compat.Final` everywhere.